tools/check_test_name.py

```diff
@@ -71,7 +71,9 @@ def is_misnamed_test_func(
         and node.name
         not in ("teardown_method", "setup_method",
                 "teardown_class", "setup_class",
-                "setup_module", "teardown_module")
+                "setup_module", "teardown_module",
+                "_test_dependency",  # array_api_compat.tests.test_no_dependencies
```

Comment:
What does this do?

Reply:
Checks that test function names start with `test` so that they are detected properly by `pytest`. I looked at what this was doing in `array_api_compat` and it looks intentional (not a typo). We could do with a better way of ignoring submodules in this script though. cc @j-bowhay @MarcoGorelli

Reply:
That's a long-standing and minor issue, so let's ignore it here (a PR is always welcome of course).